pkg/controller/defaultdomaincert/default_domain_cert_controller_test.go

```diff
@@ -0,0 +1,875 @@
+	conf := &config.Config{
+		DefaultDomainCertPath: testCertPath,
+		DefaultDomainKeyPath:  testKeyPath,
+	}
+
+	return &defaultDomainCertControllerReconciler{
+		client: client,
+		events: &record.FakeRecorder{},
+		conf:   conf,
+		store:  store,
+	}
+}
+
+func createTestDefaultDomainCertificate(name, namespace, secretName string) *approutingv1alpha1.DefaultDomainCertificate {
+	var secretPtr *string
+	if secretName != "" {
+		secretPtr = &secretName
+	}
+	// If secretName is empty string, secretPtr remains nil
+
+	return &approutingv1alpha1.DefaultDomainCertificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DefaultDomainCertificate",
+			APIVersion: approutingv1alpha1.GroupVersion.String(),
+		},
+		Spec: approutingv1alpha1.DefaultDomainCertificateSpec{
+			Target: approutingv1alpha1.DefaultDomainCertificateTarget{
+				Secret: secretPtr,
+			},
+		},
+	}
+}
+
+func TestReconcile_SuccessfulReconciliation(t *testing.T) {
+	scheme := runtime.NewScheme()
+	require.NoError(t, approutingv1alpha1.AddToScheme(scheme))
+	require.NoError(t, corev1.AddToScheme(scheme))
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", testNamespace, testSecretName)
+
+	client := fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithObjects(ddc).
+		WithStatusSubresource(ddc).
+		Build()
+
+	mockStore := newMockStore()
+	mockStore.setFileContent(testCertPath, []byte(testCertContent))
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	reconciler := createTestReconciler(client, mockStore)
+
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "test-ddc",
+			Namespace: testNamespace,
+		},
+	}
+
+	ctx := context.Background()
+	result, err := reconciler.Reconcile(ctx, req)
+
+	require.NoError(t, err)
+	assert.Equal(t, ctrl.Result{}, result)
+
+	// Verify secret was created
+	var secret corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{Name: testSecretName, Namespace: testNamespace}, &secret)
+	require.NoError(t, err)
+
+	assert.Equal(t, testSecretName, secret.Name)
+	assert.Equal(t, testNamespace, secret.Namespace)
+	assert.Equal(t, corev1.SecretTypeTLS, secret.Type)
+	assert.Equal(t, []byte(testCertContent), secret.Data["tls.crt"])
+	assert.Equal(t, []byte(testKeyContent), secret.Data["tls.key"])
+	assert.Equal(t, manifests.GetTopLevelLabels(), secret.Labels)
+
+	// Verify owner references
+	assert.Len(t, secret.OwnerReferences, 1)
+	assert.Equal(t, ddc.Name, secret.OwnerReferences[0].Name)
+	assert.Equal(t, ddc.UID, secret.OwnerReferences[0].UID)
+	assert.True(t, *secret.OwnerReferences[0].Controller)
+
+	// Verify DefaultDomainCertificate status was updated
+	require.NoError(t, client.Get(ctx, types.NamespacedName{Name: ddc.Name, Namespace: ddc.Namespace}, ddc))
+	assert.Equal(t, metav1.ConditionTrue, ddc.GetCondition(v1alpha1.DefaultDomainCertificateConditionTypeAvailable).Status)
+}
+
```

Comment:
I'm personally a bigger fan of organizing tests as a list of cases rather than separate functions but I don't feel strongly enough on that to block on this